compiler/rustc_typeck/src/check/check.rs

```diff
@@ -602,7 +604,7 @@ pub(super) fn check_opaque_for_cycles<'tcx>(
             hir::OpaqueTyOrigin::AsyncFn(..) => async_opaque_type_cycle_error(tcx, span),
             _ => opaque_type_cycle_error(tcx, def_id, span),
         }
-        Err(ErrorReported)
+        Err(ErrorReported::new_i_know_what_im_doing())
```

Comment:
`async_opaque_type_cycle_error` and `opaque_type_cycle_error` should return `ErrorReported`.